test/unit/gateways/visanet_peru_test.rb

```diff
@@ -16,20 +16,20 @@ def setup
   end
 
   def test_successful_purchase
-    @gateway.expects(:ssl_request).returns(successful_authorize_response)
-    @gateway.expects(:ssl_request).returns(successful_capture_response)
+    @gateway.expects(:ssl_request).with(:post, any_parameters).returns(successful_authorize_response)
+    @gateway.expects(:ssl_request).with(:put, any_parameters).returns(successful_capture_response)
 
     response = @gateway.purchase(@amount, @credit_card, @options)
     assert_success response
     assert_equal 'OK', response.message
 
     assert_match %r([0-9]{9}|$), response.authorization
-    assert_equal @options[:order_id], response.params['externalTransactionId']
+    assert_equal 'de9dc65c094fb4f1defddc562731af81', response.params['externalTransactionId']
     assert response.test?
   end
 
   def test_failed_purchase
-    @gateway.expects(:ssl_request).returns(failed_authorize_response_bad_card)
+    @gateway.expects(:ssl_request).with(:post, any_parameters).returns(failed_authorize_response_bad_card)
```

Comment:
I updated some, but not all, tests to make sure we would hit both `:post` and `:put` a bit.

Reply:
👍